ansible_collections/arista/avd/plugins/plugin_utils/utils/parse_validation_result.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2023-2025 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ansible.utils.display import Display
+    from pyavd_utils.validation import ValidationResult
+
+    from pyavd._utils import json_path_to_string
+
+try:
+    from pyavd._utils import json_path_to_string
+
+    HAS_PYAVD = True
+except ImportError:
+    HAS_PYAVD = False
+
+
+def parse_validation_result(validation_result: ValidationResult, hostname: str, ansible_display: Display) -> int:
+    """Parser of pyavd.load_design.ValidationResult displaying warnings and errors and returning the number of validation errors."""
+    if not HAS_PYAVD:
+        msg = "The 'arista.avd' collection requires the 'pyavd' Python library."
+        raise ImportError(msg)
+
+    for deprecation in validation_result.deprecations:
+        ansible_display.deprecated(
+            msg=f"{hostname}: {deprecation}",
```

Comment:
Otherwise it's displaying `<validation.Deprecation object at ...>`.
```suggestion
            msg=f"{hostname}: {deprecation.message}",
```
Also, Ansible appends a repeating message at the end:
```
[DEPRECATION WARNING]: DC1-BL2: The input data model 'l3leaf.defaults.structured_config.ip_name_server_groups[0].name_servers' is deprecated and will be removed in AVD version 7.0.0. This feature will be removed from collection 'arista.avd' version 7.0.0.
```